### PR TITLE
The Big Merge™

### PR DESCRIPTION
**Pull request by DeltaDesigns**
(I'm sorry if this causes a headache)
- New Entities tab in Activity View (map and activity entities) (Activity entities for WQ+)
- Map background entities (Skybox stuff) (WQ+)
- Map decals (textures)
- Basic map lights (point lights)
- Fix Shadowkeep terrain
- Fix Shadowkeep entity external materials
- Source 2 shader updates

I've tested enough to confirm that the big things wont crash, though I haven't tested Beyond Light for the changes/additions

### CODE REVIEW

File: Tiger/Schema/Model/VertexBuffer.cs [hunk 1]
```diff
@@ -410,22 +410,29 @@ private bool ReadVertexDataType6(TigerReader handle, DynamicMeshPart dynamicPart
                     handle.BaseStream.Seek(chunkIndex * 0x20 + (vertexIndex % 8) * 4, SeekOrigin.Begin);
                     vw.WeightIndices = new IntVector4(handle.ReadByte(), handle.ReadByte(), 0, 0);
                     vw.WeightValues = new IntVector4(handle.ReadByte(), handle.ReadByte(), 0, 0);
-                    Debug.Assert(vw.WeightValues.X + vw.WeightValues.Y == 255);
```
Comment (MontagueM) [inline Tiger/Schema/Model/VertexBuffer.cs:413]: why remove this assert? if it fails, the weight calculation is definitely wrong

File: Tiger/Schema/Model/VertexBuffer.cs [hunk 2]
```diff
@@ -381,8 +381,8 @@ private bool ReadVertexDataType6(TigerReader handle, DynamicMeshPart dynamicPart
                  */
 
                 // new code vvv
-                VertexWeight vw = new();
-                short w = (short)dynamicPart.VertexPositions[dynamicPart.VertexIndexMap[vertexIndex]].W;
+                VertexWeight vw = new VertexWeight();
+                short w = (short)dynamicPart.VertexPositions[dynamicPart.VertexIndices.IndexOf(vertexIndex)].W;
```
Comment (MontagueM) [inline Tiger/Schema/Model/VertexBuffer.cs:385]: this replacement should be identical, the VertexIndexMap is done as a cache as IndexOf is slow

File: Tiger/Schema/Shaders/TextureExtractor.cs [hunk 3]
```diff
@@ -28,9 +28,9 @@ public static bool SaveTextureToFile(string savePath, ScratchImage scratchImage,
                     break;
                 case TextureExportFormat.DDS_BGRA_BC7_DX10:
                     if (TexHelper.Instance.IsSRGB(scratchImage.GetMetadata().Format))
-                        scratchImage = scratchImage.Compress(DXGI_FORMAT.BC7_UNORM_SRGB, TEX_COMPRESS_FLAGS.SRGB, 0);
+                        scratchImage = scratchImage.Compress(DXGI_FORMAT.BC3_UNORM_SRGB, TEX_COMPRESS_FLAGS.SRGB, 0);
```
Comment (MontagueM) [inline Tiger/Schema/Shaders/TextureExtractor.cs:31]: why BC3 change? if fine then can you also pls change the texture format enum which states it's BC7

File: Tiger/Schema/Test.cs
Comment (MontagueM) [inline Tiger/Schema/Test.cs:1]: could be renamed to Lights presumably

Comment (MontagueM) [inline Tiger/Schema/Test.cs:1]: worth deleting or nah

Comment (DeltaDesigns) [reply Tiger/Schema/Test.cs:1]: yeah

**Review by MontagueM — APPROVED** [review]

**Pull request merged**